cmd/gateway-gcs.go

```diff
@@ -188,14 +191,56 @@ func gcsToObjectError(err error, params ...string) error {
 // gcsProjectIDRegex defines a valid gcs project id format
 var gcsProjectIDRegex = regexp.MustCompile("^[a-z][a-z0-9-]{5,29}$")
 
-// isValidGCSProjectId - checks if a given project id is valid or not.
-// Project IDs must start with a lowercase letter and can have lowercase
-// ASCII letters, digits or hyphens. Project IDs must be between 6 and 30 characters.
+// gcsProjectIDValidFormat - checks if a given project id format is valid or not.
+// Project IDs must start with a lowercase letter and can have lowercase ASCII letters,
+// digits or hyphens. Project IDs must be between 6 and 30 characters.
 // Ref: https://cloud.google.com/resource-manager/reference/rest/v1/projects#Project (projectId section)
-func isValidGCSProjectID(projectID string) bool {
+func gcsProjectIDValidFormat(projectID string) bool {
+	// Checking projectID format
 	return gcsProjectIDRegex.MatchString(projectID)
 }
 
+// gcsProjectIDExists - checks if the project ID does really exist using resource manager API.
+func gcsProjectIDExists(ctx context.Context, projectID string) error {
```

Comment:
Usually, functions that check for existence return `bool`. E.g, https://golang.org/pkg/os/#IsExist.
If returning `error` makes more sense here, we can call this function `validateGCSProjectID`.